sympy/printing/pretty/tests/test_pretty.py

```diff
@@ -6346,19 +6346,19 @@ def test_issue_8292():
     e = sympify('((x+x**4)/(x-1))-(2*(x-1)**4/(x-1)**4)', evaluate=False)
     ucode_str = \
 """\
-           4    4    \n\
-  2⋅(x - 1)    x  + x\n\
-- ────────── + ──────\n\
-          4    x - 1 \n\
-   (x - 1)           \
+              4     4     \n\
+  2⋅(x + -1⋅1)     x  + x \n\
+- ───────────── + ────────\n\
+             4    x + -1⋅1\n\
+   (x + -1⋅1)             \
```

Comment:
I have little experience with the printer, but as a user, I wouldn't want to see the `-1⋅1`. Preferably, the output should look pretty close to the input. Do you think that there's a way to revert this without changing the parser?

Reply:
Hm, I see your point. However, in that test, the input was generated with `sympify`:

https://github.com/sympy/sympy/blob/78f99c02c5b79eeae6017771056f21a136f58d84/sympy/printing/pretty/tests/test_pretty.py#L6344-L6364

which actually returns this:

```python
-2*(x + (-1)*1)**4/(x + (-1)*1)**4 + (x**4 + x)/(x + (-1)*1)
```

So technically `pretty` would just be printing exactly what it gets from `sympify`. Is this desired? If not, I'm kind of stuck as to how to prevent that and fix the linked issue at the same time.

Reply:
I suspect the problem is will all negative numbers. That is, `-2` becomes `(-1)*2` and `-3.14` becomes `(-1)*3.14`.

I guess a choice would need to be made about it. I think we have 2 options:
1. Commit the PR with this change. (Users that "translate" SymPy expressions to LaTeX or others would need manually change negative expressions).
2. Add an if statement in the `mul` printer to check if we have `(-1)*1` in which case it would evaluate itself and print `-1`. Users that want to see the `*1` would need to manually add it in.

We might need to wait for someone more knowledgeable than me on the matter.

Reply:
Previously Mul would absorb identities even when `evaluate=False` meaning that there was no way to create e.g. `Mul(1, 2, evaluate=False)`. I changed that in #19450. That has since thrown up various problems around the code base: previously such unevaluated Muls didn't exist.

While working on that I found that it was difficult to debug problems when unevaluated Muls displayed as ordinary rationals e.g. https://github.com/sympy/sympy/pull/19450#issuecomment-636060075. I added a check for an unevaluated Mul and special cased it to print differently. I don't mind that being changed but I think it's important to be able to "see" an unevaluated Mul.

My take on the printing of Muls is that if you have something like `Mul(1, -1)` then it must have been created explicitly using evaluate=False or something else. Either:

1. You had a reason for wanting to do that
2. You didn't actually want to have an unevaluated Mul in which case it is good if the printing code shows you what you have so you can fix your code not to produce the unevaluated Mul.

Regarding point 1. bear in mind that one of the main reason users create unevaluated Muls is to try and control the printed output e.g.:
https://stackoverflow.com/questions/65979823/keep-fractions-as-factors

Regarding point 2. bear in mind that:
```python
In [1]: Mul(1, 1, evaluate=False) == 1
Out[1]: False

In [2]: Mul(1, 1) == 1
Out[2]: True
```
Using `Mul(1, 1)` in place of `Integer(1)` can break all sorts of things and if `Mul(1, 1)` displays the same as `Integer(1)` then that's really confusing and hard to debug.

Reply:
@oscarbenjamin So do you think the change is acceptable since it directly translates what `sympify` produces? Or should it do something else?

Reply:
I'm not sure what should be done here. I haven't had time to go through what the issue here is. I just wanted to point out that printing unevaluated Muls as they look after evaluation is problematic.

Reply:
Got it. Thanks for elaborating!